internal/database/autocontact_test.go

```diff
@@ -0,0 +1,152 @@
+package database
+
+import (
+	"log/slog"
+	"os"
+	"testing"
+	"time"
+
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+)
+
+func TestSSHRestart(t *testing.T) {
+	// Setup
+	logger := slog.Default()
+	sshConfig := SSHConfig{
+		User:    "testuser",
+		Keypath: "/path/to/your/test/private/key",
+	}
+
+	err := sshRestart("invalid.remote.address", logger, sshConfig)
+	if err == nil {
+		t.Error("Expected sshRestart to fail due to invalid setup, but it did not")
+	} else {
+		t.Log("SSHRestart attempted as expected:", err)
+	}
+
+}
+
+func TestSSHRestart_UnreadableKey(t *testing.T) {
+	logger := slog.Default()
+	sshConfig := SSHConfig{
+		User:    "testuser",
+		Keypath: "/path/to/nonexistent/key", // Intentionally wrong to trigger the read error
+	}
+
+	err := sshRestart("localhost", logger, sshConfig)
+	if err == nil {
+		t.Error("Expected an error due to unreadable key, but got none")
+	} else {
+		t.Log("Received expected error:", err)
+	}
+}
+
+func TestSSHRestart_UnparsableKey(t *testing.T) {
+	logger := slog.Default()
+	// Create a temp file with invalid SSH key content
+	tmpFile, err := os.CreateTemp("", "invalid-ssh-key-*")
+	if err != nil {
+		t.Fatalf("Failed to create temp file: %v", err)
+	}
+	defer os.Remove(tmpFile.Name()) // clean up
+
+	_, err = tmpFile.WriteString("not-a-valid-key")
+	if err != nil {
+		t.Fatalf("Failed to write to temp file: %v", err)
+	}
+	tmpFile.Close()
+
+	sshConfig := SSHConfig{
+		User:    "testuser",
+		Keypath: tmpFile.Name(), // Use the temp file with invalid content
+	}
+
+	err = sshRestart("localhost", logger, sshConfig)
+	if err == nil {
+		t.Error("Expected an error due to unparsable key, but got none")
+	} else {
+		t.Log("Received expected error:", err)
+	}
+}
+
+func TestPing(t *testing.T) {
+	logger := slog.Default()
+
+	success := ping("192.0.2.1", logger)
```

Comment:
NIT: Can you add a comment that this is designated as non-routable by https://datatracker.ietf.org/doc/html/rfc5737